### PR TITLE
cleanup gripper agent

### PR DESCRIPTION
**Pull request by jlashner**
A few changes to the gripper-agent to address some things mentioned by Brian in the PR comments and to make it more compatible with the [PRU monitor](https://github.com/simonsobs/sobonelib/pull/2) sobonelib changes.

This needs testing

### CODE REVIEW

**Review by bbixler500 — APPROVED** [review]

File: socs/agents/hwp_supervisor/agent.py [hunk 1]
```diff
@@ -259,6 +265,28 @@ def update_ups_state(self, op):
...
+            ``acq`` process
+        """
+        iboot_keymap = {
+            'iboot_outlet1_state': (f'outletStatus_{self.iboot_outlet1}', 'status'),
+            'iboot_outlet2_state': (f'outletStatus_{self.iboot_outlet2}', 'status'),
+        }
+
+        if op['status'] != 'ok':
+            for k in ups_keymap:
+                setattr(self, k, None)
```
Comment (jlashner) [inline socs/agents/hwp_supervisor/agent.py:285]: This should be `iboot_keymap` instead of `ups_keymap`

File: socs/agents/hwp_supervisor/agent.py [hunk 2]
```diff
@@ -688,10 +718,15 @@ def __init__(self, agent, args):
         self.hwp_pid_id = args.hwp_pid_id
         self.ups_id = args.ups_id
 
+        self.iboot_id = args.iboot_id
+        self.iboot_outlets = eval(args.iboot_outlets)
```
Comment (BrianJKoopman) [inline socs/agents/hwp_supervisor/agent.py:722]: What's in `args.iboot_outlets`? I'm wondering why the `eval()` is needed.

File: docs/conf.py [hunk 3]
```diff
@@ -104,6 +104,8 @@
                         'pfeiffer_vacuum_protocol',
                         'pfeiffer_vacuum_protocol.pfeiffer_vacuum_protocol',
                         'pfeiffer_tc400_driver',
+                        'socs.agents.hwp_supervisor.agent',
```
Comment (BrianJKoopman) [inline docs/conf.py:107]: I think this mock is preventing the auto-generated docs from populating in the API section. See: https://socs--487.org.readthedocs.build/en/487/agents/hwp_supervisor_agent.html#agent-api

File: socs/agents/hwp_supervisor/agent.py [hunk 4]
```diff
@@ -986,7 +1026,8 @@ def make_parser(parser=None):
     pgroup.add_argument('--ups-minutes-remaining-thresh', type=float,
                         help="Threshold for UPS minutes remaining before a "
                              "shutdown is triggered")
-
+    pgroup.add_argument('--iboot-id', help="Instance ID for IBoot-PDU agent")
+    pgroup.add_argument('--iboot-outlets', help="IBoot-PDU outlets connected  to bias power")
```
Comment (BrianJKoopman) [inline socs/agents/hwp_supervisor/agent.py:1030]: Add to example SCF block on the docs page.

**Review by BrianJKoopman — CHANGES_REQUESTED** [review]
Thanks for the structure clean up effort @jlashner! I mostly have documentation requests.

I feel pretty strongly about the need to have examples of the `session.data` in the docstrings, but the structure here of passing whatever comes back from the `GripperServer` in `sobonelib` straight to `session.data` makes that perhaps a bit awkward since changes in `sobonelib` might result in changes in the returned data structure that are documented here.

File: docs/conf.py (hunk 3 above)
Comment (jlashner) [reply docs/conf.py:107]: Yep I definitely added that as a temp fix before the supervisor agent was merged in. Thanks!

File: socs/agents/hwp_gripper/agent.py [hunk 5]
```diff
@@ -198,6 +239,14 @@ def inp(self, session, params=None):
...
         tells you whether the windows software has detected that the actuator
         has been homed.
+
+        Notes:
+            The most recent data collected is stored in session data in the
+            structure:
+
+                >>> response.session['response']
+                {'result': True,
+                 'log': []}
```
Comment (BrianJKoopman) [inline socs/agents/hwp_gripper/agent.py:250]: Does any logging happen for the INP command? Or does this always come back empty?

File: socs/agents/hwp_gripper/agent.py [hunk 6]
```diff
@@ -0,0 +1,700 @@
...
+            time.sleep(sleep_time)
+
+    def _stop_monitor_state(self, session, params=None):
+        session.set_status('stopping')
+        return True, "Requesting monitor_state process to stop"
+
+    @ocs_agent.param('no_data_warn_time', default=60, type=float)
+    @ocs_agent.param('no_data_shutdown_time', default=300, type=float)
+    def monitor_supervisor(self, session, params=None):
+        """monitor_supervisor()
```
Comment (BrianJKoopman) [inline socs/agents/hwp_gripper/agent.py:573]: `monitor_supervisor()` -> `monitor_supervisor(no_data_warn_time=60, no_data_shutdown_time=300)`

File: socs/agents/hwp_gripper/agent.py (hunk 5 above)
Comment (bbixler500) [reply socs/agents/hwp_gripper/agent.py:250]: It always comes back empty

**Review by BrianJKoopman — CHANGES_REQUESTED** [review]
Thanks for addressing all those comments @bbixler500. There's still one missing `session.data` example (I left the comment unresolved in the last review.) And also there's the mock that's still causing issues build the docs. Lastly, there are still a failing test, it looks related to the ibootbar states.

File: docs/conf.py (hunk 3 above)
Comment (BrianJKoopman) [reply docs/conf.py]: Great, both HWP supervisor agent mocks are now removed. I don't think either was really needed anymore.

**Comment by BrianJKoopman** [comment]
Let me know if I can help with the tests. Everything else looks good.

**Review by BrianJKoopman — APPROVED** [review]
Thanks for fixing the tests!

**Pull request merged**